### PR TITLE
feat(dashboard,core-flows,medusa): update fulfillment flows

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- update fulfillment create/cancel flows to work with latest API
- cancel order action

**Fix**
- fix cancel fulfillment endpoint to accept params correctly
- fix inventory adjustment step in the cancel workflow if the item's variant doesn't have `manage_inventory`
- make cancel order endpoint POST

---

CLOSES CORE-2234

### CODE REVIEW

**Comment by olivermrbl** [comment]
Getting this error on creating fulfillment right now:
CleanShot 2024-06-04 at 10 29 40

I think this will be fixed by #7592, as this is because there are no reservations for the variant, because we have `manage_inventory=false`. 

Can you confirm?

**Comment by fPolic** [comment]
@olivermrbl - exactly, this is fixed with the https://github.com/medusajs/medusa/pull/7594

File: packages/admin-next/dashboard/src/components/table/table-cells/order/fulfillment-status-cell/fulfillment-status-cell.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import type { FulfillmentStatus } from "@medusajs/medusa"
+import { FulfillmentStatus } from "@medusajs/medusa"
```
Comment (carlos-r-l-rodrigues) [inline packages/admin-next/dashboard/src/components/table/table-cells/order/fulfillment-status-cell/fulfillment-status-cell.tsx]: The admin should not import `@medusa/medusa`, should it?

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM. just 1 import issue

File: packages/admin-next/dashboard/src/components/table/table-cells/order/fulfillment-status-cell/fulfillment-status-cell.tsx (hunk 1 above)
Comment (fPolic) [reply packages/admin-next/dashboard/src/components/table/table-cells/order/fulfillment-status-cell/fulfillment-status-cell.tsx]: Yes, good catch this was wrong autoimport

File: packages/admin-next/dashboard/src/hooks/api/orders.ts [hunk 2]
```diff
@@ -38,3 +45,57 @@ export const useOrders = (
 
   return { ...data, ...rest }
 }
+
+export const useCreateOrderFulfillment = (
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/hooks/api/orders.ts]: thought: Should we migrate the client endpoints to use the SDK?

File: packages/admin-next/dashboard/src/lib/client/orders.ts [hunk 3]
```diff
@@ -1,5 +1,10 @@
-import { getRequest } from "./common"
-import { OrderListRes, OrderRes } from "../../types/api-responses"
+import { getRequest, postRequest } from "./common"
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/lib/client/orders.ts]: I guess we can remove this now right?

File: packages/core/js-sdk/src/admin/order.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  AdminCancelOrderFulfillment,
+  FindParams,
+  HttpTypes,
+  PaginatedResponse,
+  SelectParams,
+} from "@medusajs/types"
+import { Client } from "../client"
+import { ClientHeaders } from "../types"
+
+export class Order {
```
Comment (olivermrbl) [inline packages/core/js-sdk/src/admin/order.ts:11]: Nice, think we should create types for this, but to not extend the scope of the PR further, we could save that for a follow-up PR

**Review by olivermrbl — APPROVED** [review]
LGTM, just one comment

**Pull request merged**